vizro-core/src/vizro/actions/_action_loop/_build_action_loop_callbacks.py

```diff
@@ -23,10 +23,28 @@ def _build_action_loop_callbacks() -> None:
     actions_chains = _get_actions_chains_on_registered_pages()
     actions = _get_actions_on_registered_pages()
 
-    gateway_triggers: List[Input] = []
+    if not actions_chains:
+        return
+
+    gateway_inputs: Dict[str, Any] = {
+        "gateway_triggers": [],
+        "cycle_breaker_div": Input("cycle_breaker_div", "n_clicks"),
+        "remaining_actions": State("remaining_actions", "data"),
+
+    }
+    gateway_outputs: Dict[str, Any] = {
+        "action_triggers": [Output({"type": "action_trigger", "action_name": action.id}, "data") for action in actions],
+        "remaining_actions": Output("remaining_actions", "data"),
+    }
+
     for actions_chain in actions_chains:
-        # TODO: Potentially convert to clientside callback
-        @callback(
+        # Callback that enables gateway callback to work in the multiple page app
+        clientside_callback(
```

Comment:
By the way, is there any good way of testing these clientside callbacks? It seems more difficult given that the function is wrapped inside and one cannot refactor it out. This is how we currently test our callbacks: https://stackoverflow.com/questions/67638178/how-to-write-testcases-for-plotly-dash-callback-function

Is there anything similar we could do?

Reply:
Let's continue this conversation in the thread [here](https://github.com/mckinsey/vizro/pull/34/files#r1324696236).